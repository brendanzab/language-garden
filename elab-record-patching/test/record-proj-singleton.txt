let _ : fun (A : Type) (x : A) -> A :=
  fun A x =>
    let record : { x : A } [= { x }] := { x };
    --                                  ^^^^^ Elaborates to `#sing-intro`
    --                                        when checking `{ x }` against `{ x : A } [= { x }]`

    record.x;
--  ^^^^^^ Elaborates to `{ x }`

Type
