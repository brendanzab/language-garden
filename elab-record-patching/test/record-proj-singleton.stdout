<input> : Type :=
  let _ :=
    (fun A x :=
       let record := #sing-intro : { x : A } [= { x := x }]; { x := x }.x) :
      fun (A : Type) (x : A) -> A;
  Type
